### PR TITLE
Support linestring

### PR DESCRIPTION
**Pull request by akacarlyann**
Adds class for LinestringViz with geojson and vector data source support.  Line color and width are data-driven styled.  Addresses #4, #81.

Still working on:
- [x] legend refinement
- [x] default color/line width test (make sure consistent with latest CircleViz and ChoroplethViz)
- [ ] bugfix for match-type color on vector linestring example
- [x] clean up example linestring-viz.ipynb

### CODE REVIEW

**Comment by akacarlyann** [comment]
@ryanbaumann Ready for you to take a first look.

Bug mentioned above is actually fixed with a refinement to the __color_map__ function (when I wrote it I assumed match-type would entail non-numeric stops, given numeric color stops and ```color_function_type = 'match'```, __color_map__ is interpolating rather than returning the default color).  Fix would be adding an argument to __color_map__ to override this behavior for 'match' __color_function_type__ and numeric stops.  I can open a separate PR for that -- thoughts?

**Comment by ryanbaumann** [comment]
New PR for the bug fix on color_map @akacarlyann 👍 .

This is amazing.  Taking a look now.

**Comment by akacarlyann** [comment]
Perfect! I knew you'd have something in the wings for this.

File: mapboxgl/templates/linestring.html [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            "type": "symbol",
+            "layout": {
+                {% if labelProperty %}
+                "text-field": "{{ labelProperty }}",
+                {% endif %}
+                "text-size" : generateInterpolateExpression('zoom', [[0,8],[22,16]] ),
+                "text-offset": [0,-1]
+            },
+            "paint": {
+                "text-halo-color": "white",
```
Comment (ryanbaumann) [inline mapboxgl/templates/linestring.html]: Make text-color, halo-color, halo-width, and text-size adjustable to math other visuals labels.  https://github.com/mapbox/mapboxgl-jupyter/blob/master/mapboxgl/templates/circle.html#L38

File: mapboxgl/templates/vector_linestring.html [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    map.addLayer({
+        "id": "linestring-label",
+        "source": "vector-data",
+        "source-layer": "{{ vectorLayer }}",
+        "type": "symbol",
+        "layout": {
+            {% if labelProperty %}
+            "text-field": "{{ labelProperty }}",
+            {% endif %}
+            "text-size" : generateInterpolateExpression('zoom', [[0,8],[22,16]] ),
```
Comment (ryanbaumann) [inline mapboxgl/templates/vector_linestring.html]: Likewise in the vector linestring layer - add style modifiers for text-color, text-size, halo-color, halo-width

Comment (akacarlyann) [reply mapboxgl/templates/vector_linestring.html]: Thanks!  Added these in 3e9fac4.  Looks like I'll need to update ChoroplethViz too in a separate issue.

**Comment by ryanbaumann** [comment]
Nice @akacarlyann!  Add the new lable style property options to the Markdown Docs, then 🚢 !  So excited to get this one into python dev's hands.

**Comment by akacarlyann** [comment]
@ryanbaumann Added __LinestringViz__ to __viz.md__ in 9bf8311.

**Pull request merged**